tests/k8st/tests/test_ipip_spoofing.py

```diff
@@ -0,0 +1,229 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import subprocess
+from kubernetes import client
+
+from tests.k8st.test_base import TestBase
+from tests.k8st.utils.utils import \
+    retry_until_success, \
+    DiagsCollector, \
+    calicoctl, \
+    calicoctl_apply_dict, \
+    kubectl, \
+    generate_unique_id
+
+_log = logging.getLogger(__name__)
+
+
+class ConnectionError(Exception):
+    pass
+
+
+class TestIPIPSpoof(TestBase):
+    def setUp(self):
+        TestBase.setUp(self)
+        self.ns_name = generate_unique_id(5, prefix="spoof")
+        self.create_namespace(self.ns_name)
+        # Create two client pods that live for the duration of the
+        # test.  We will use 'kubectl exec' to try sending/receiving
+        # from these at particular times.
+        #
+        # We do it this way because it takes a
+        # relatively long time (7 seconds?) in this test setup for
+        # Calico routing and policy to be set up correctly for a newly
+        # created pod.
+        kubectl("run --generator=run-pod/v1 "
+                "access "
+                "-n %s "
+                "--image busybox "
+                "--overrides='{\"spec\": {\"nodeName\":\"kube-node-1\"}}' "
+                "--command /bin/sh -- -c \"nc -l -u -p 5000 &> /root/snoop.txt\"" % self.ns_name)
+        kubectl("run --generator=run-pod/v1 "
+                "scapy "
+                "-n %s "
+                "--image ehlers/scapy "
+                "--overrides='{\"spec\": {\"nodeName\":\"kube-node-2\"}}' "
+                "--command /bin/sleep -- 3600" % self.ns_name)
+
+        kubectl("wait --timeout=2m --for=condition=ready" +
+                " pod/scapy -n %s" % self.ns_name)
+        kubectl("wait --timeout=2m --for=condition=ready" +
+                " pod/access -n %s" % self.ns_name)
+
+    def tearDown(self):
+        # Delete deployment
+        self.delete_and_confirm(self.ns_name, "ns")
+        # Change pool to use IPIP
+        default_pool = json.loads(calicoctl("get ippool default-ipv4-ippool -o json"))
+        default_pool["spec"]["vxlanMode"] = "Never"
+        default_pool["spec"]["ipipMode"] = "Always"
+        calicoctl_apply_dict(default_pool)
+        # restart calico-nodes
+        kubectl("delete po -n kube-system -l k8s-app=calico-node")
+        kubectl("wait --timeout=2m --for=condition=ready" +
+                " pods -l k8s-app=calico-node -n kube-system")
+
+    def test_simple_ipip_spoof(self):
+        with DiagsCollector():
+            # Change pool to use IPIP
+            default_pool = json.loads(calicoctl("get ippool default-ipv4-ippool -o json"))
+            default_pool["spec"]["vxlanMode"] = "Never"
```

Comment:
Just a nit, but since IPIP is the default it might worth checking if we need to apply this first.